internal/explore/purl.go

```diff
@@ -81,7 +81,11 @@ func (p *purl) url(repo string) (string, error) {
 		}
 		return fmt.Sprintf("/%s%s%s%s", h, repository, delim, p.version), nil
 	case "github":
-		return fmt.Sprintf("https://github.com/%s/%s/tree/%s", p.namespace, p.name, p.version), nil
+		if p.subpath == "" {
+			return fmt.Sprintf("https://github.com/%s/%s/tree/%s", p.namespace, p.name, p.version), nil
+		} else {
+			return fmt.Sprintf("https://github.com/%s/%s/blob/%s/%s", p.namespace, p.name, p.version, p.subpath), nil
```

Comment:
Tree will technically redirect to blob when appropriate, so you could take advantage of that and use `path.Join` if you wanted to avoid the conditional/branching 👀

Reply:
ie https://github.com/wolfi-dev/os/tree/92940c6d5e40c38f076ad57ec660e765f8ef2e7a/openssl.yaml and https://github.com/wolfi-dev/os/blob/92940c6d5e40c38f076ad57ec660e765f8ef2e7a/openssl.yaml end up in the same place